plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java

```diff
@@ -35,196 +57,445 @@
+			.applyTo(serviceText);
+		ValueBindingBuilder
+			.bind(WidgetProperties.text(SWT.Modify).observe(serviceText))
+			.to(BeanProperties.value(OpenShiftServerEditorModel.PROPERTY_SERVICE).observe(model))
+			.converting(new Converter(IService.class, String.class) {
+				
+				@Override
+				public Object convert(Object fromObject) {
+					if (!(fromObject instanceof IService)) {
+						return null;
+					};
+					return ((IService) fromObject).getName();
+				}
+			})
+			.in(dbc);
+
+		Button selectServiceButton = new Button(parent, SWT.PUSH);
+		selectServiceButton.setText("Select...");
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.CENTER).hint(100, SWT.DEFAULT)
+				.applyTo(selectServiceButton);
+		selectServiceButton.addSelectionListener(onSelectService(model, selectServiceButton.getShell()));
 	}
 
-	private void updateWidgetsFromWorkingCopy() {
-//		connectionText.setText(createConnectionLabel(OpenShiftServerUtils.getConnectionUrl(server)));
-//		buildConfigText.setText("");
-//		projectNameText.setText(StringUtils.null2emptyString(OpenShiftServerUtils.getOpenShiftProject(server)));
+	private SelectionListener onSelectService(OpenShiftServerEditorModel model, final Shell shell) {
+		return new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+
+				
+				SelectServiceWizard selectServiceDialog = 
+						new SelectServiceWizard(NLS.bind("Select a service that your server adapter {0} will publish to.", 
+								input.getServer().getName()), model.getService(), model.getConnection());
+				if (WizardUtils.openWizardDialog(selectServiceDialog, shell) 
+						== Dialog.OK) {
+					model.setService(selectServiceDialog.getService());
+				}
+			}
+		};
+	}
+
+	private void loadResources(Composite container, OpenShiftServerEditorModel model) {
+		IServerWorkingCopy server = input.getServer();
+
+		final Connection connection = OpenShiftServerUtils.getConnection(server);
+		final IProject deployProject = OpenShiftServerUtils.getDeployProject(server);
+
+		Cursor busyCursor = new Cursor(container.getDisplay(), SWT.CURSOR_WAIT);
+		new JobChainBuilder(
+				new DisableAllWidgetsJob(true, container, busyCursor))
+				.runWhenDone(
+						new Job("Loading projects...") {
+
+							@Override
+							protected IStatus run(IProgressMonitor monitor) {
+								model.loadResources();
+								return Status.OK_STATUS;
+							}
+						})
+				.runWhenDone(
+						new Job("Setting connection, deploy project...") {
+
+							@Override
+							protected IStatus run(IProgressMonitor monitor) {
+								model.setService(OpenShiftServerUtils.getService(server));
+								if (deployProject != null) {
+									model.setDeployProject(deployProject);
+								}
+								if (connection != null) {
+									model.setConnection(connection);
+								}
+								return Status.OK_STATUS;
+							}
+						})
+				.runWhenDone(new DisableAllWidgetsJob(true, container, false, busyCursor))
+				.schedule();
```

Comment:
it should be true if we editing was implemented (editing was not implemented since it's non-obvious for a server editor to turn dirty - and allow saving - when editing a project and NOT the server). I therefore turned this to false on purpose to disable editing for now. But of course, once editing is supported the "true" will become a "false" so that the widgets may be used to edit values.